pcsx2/Recording/InputRecordingFile.cpp

```diff
@@ -158,29 +158,19 @@ bool InputRecordingFile::OpenExisting(const wxString path)
 
 bool InputRecordingFile::ReadKeyBuffer(u8 &result, const uint &frame, const uint port, const uint bufIndex)
 {
-	if (recordingFile == nullptr)
-	{
-		return false;
-	}
-
 	long seek = getRecordingBlockSeekPoint(frame) + controllerInputBytes * port + bufIndex;
-	if (fseek(recordingFile, seek, SEEK_SET) != 0 || fread(&result, 1, 1, recordingFile) != 1)
-	{
-		return false;
-	}
-
-	return true;
+	return fseek(recordingFile, seek, SEEK_SET) == 0 && fread(&result, 1, 1, recordingFile) == 1;
 }
 
 void InputRecordingFile::SetTotalFrames(long frame)
 {
-	if (recordingFile == nullptr || totalFrames >= frame)
+	if (totalFrames < frame)
 	{
-		return;
+		totalFrames = frame;
+		fseek(recordingFile, seekpointTotalFrames, SEEK_SET);
+		fwrite(&totalFrames, 4, 1, recordingFile);
 	}
-	totalFrames = frame;
-	fseek(recordingFile, seekpointTotalFrames, SEEK_SET);
-	fwrite(&totalFrames, 4, 1, recordingFile);
+	fflush(recordingFile);
```

Comment:
I'm not personally very familiar with C file management.  Why do we need the `fflush` here, and removed it from `WriteKeyBuffer`?  Otherwise this looks great as well, but same question around removing the `nullptr` check